### ISSUE TITLE
I Found Some Code for Adding a User Option to Enable Full URLs

### ISSUE THREAD

**Issue by CharmCityCrab** [fork-maintainers/iceraven-browser#1]
Honestly, as a non-coder, I had a bit of trouble figuring out how to add a pull request or a commit for this (One of which I gather is where I should really be putting it).  Hopefully given the new and relatively informal nature of this fork, @interfect won't mind me advising him of this the only way I know how to.

Someone has created a commit with a patch to add a user option that would enable full URLs on Fenix (official).  I suspect that it may not make it through to their official release (I could be wrong).  However, it seems perfect for this project.

Here is a link to the patch:

https://github.com/hakkikaancaliskan/fenix/commit/f697183677e319bd93fc3de2e97f6a6437ea7478

I don't know the gentleman who wrote the code and can't read the code, so I'm relying on more knowledgeable people to figure out if it does what it says it does and only what it says it does.

I'm also not sure what the etiquette is here for using someone else's code, but he submitted it to a project this is a close fork of, so it would seem to me that it implicitly or explicitly must be be intended to have the same open-source licenses that this project has, and should thus be something this project could use if desired (If not, he could be contacted and asked for permission, I suppose).

A PDF file is attached in case the original link is inaccessible for whatever reason (It's just the front page of the commit).
[FenixURLCode - Copy.pdf](https://github.com/interfect/fenix/files/5070719/FenixURLCode.-.Copy.pdf)

**Comment by ghost** [comment]
@CharmCityCrab The contributor @hakkikaancaliskan has been contributing to the Fenix repository. I am not programmer but from what I understand they also opened a Pull request (PR) on `https://github.com/mozilla-mobile/fenix/pull/13650` as well so all that remains is that the developers approve it. I am not a programmer so I also don't know how to read the code they have published but I can at least assure that if you search their contributions in the Fenix repository you can see that they have contributed in the past as well and I have seen that they get approved as well. They also have a 'Contributor' badge you can see in the Fenix repository.  

Currently the PR: `https://github.com/mozilla-mobile/fenix/pull/13650` is pending for UX approval/feedback (see the badge indicating the same on the PR itself). And if they approve it; it might get merged or might not, I don't know what might happen next. Just hope that it is something that gets approved for the sake of protecting user' choice.



Best of luck for all the Fenix users who believe in customizability from my side.

**Comment by ghost** [comment]
Hi, @CharmCityCrab how are you doing my friend? I hope you're fine, please don't get upset about this whole situation, okay? things like these happen...

After I made some comments; `https://github.com/mozilla-mobile/fenix/issues/12811#issuecomment-673982616`, `https://github.com/mozilla-mobile/fenix/issues/12811#issuecomment-674332872`;

I think we all somewhat made it. 

They have renamed the issue's title to: "An option to show the full website's address and protocol" as you and I and others suggested.

There's a pull request pending for the approval already on their repository for this and all that remains is that they approve it and in the spirit of browser customizability; provide the customizability of the browser in the hands of the users using that browser and if they still refuse to do so then I think at least we should respect their choice and move on.

**Comment by CharmCityCrab** [comment]
Hi @FrostedIce339 .  I appreciate your asking after my well-being and your advice.

I noted the questions in the Mozilla-Fenix GitHub about whether or not I have been banned. I can confirm that I have been banned from the Mozilla Github (I can continue to post here on this Github because it is for a fork and not under Mozilla's jurisdiction.).

The stated reason for the ban in the ban email I received was the thread about restoring full URLs (With no further explanation), but it came within a hours after I, on Reddit (Note: Not GitHub), criticized what I viewed as some inappropriate gestures, language, and tone on Twitter by two Fenix developers (Primarily one) that appeared to be aimed generically at Firefox users who expressed concern about or disagreed with how they were handling the Fenix transition.  Those posts were made by them in public with accounts in which they noted their jobs in their profile and acknowledged them in their tweet dialog (I really do want to emphasize the point that these were public posts they made with their real names recently in Twitter accounts followed by many Firefox users. I was not "doxxing" anyone, nor would I.  I also was not making any private conversations or pictures public or tying things from people who wanted to keep their Twitter accounts separate from their professional lives together.).  It is my view that I was banned on Mozilla's GitHub for essentially whistleblowing on Reddit, though it may not be possible to prove that one way or another- I just think it's likely (With the rumor that only six Mozilla people are assigned to Fenix, and that Firefox's desktop browser uses Bugzilla instead of GitHub, it seems very possible that the same six Fenix developers who maintain the browser have the keys [figuratively speaking] to the Mozilla mobile browser GitHub.).

The reason I didn't reply to your first post in this thread was because I was trying to lay low and not risk having them exclude a feature from a browser with worldwide marketshare essentially because they don't like me (I would normally count on what I would assume to be the typically innate professionalism of anyone in that position for that, but those Twitter posts were not very professional, nor was the retaliatory banning, so it was an open question as to whether or not my involvement would hurt the chances of the feature being included or not.).

However, the main developer has since posted Twitter comments directly aimed at me and my Reddit posts, accusing me of various things in a very unprofessional attempt to evoke sympathy and raise a virtual mob against me, so I figure at this point whatever she is going to do is what she is going to do.

I would probably have accepted an unbanning and a public apology, but she chose to "call me out" instead, so to speak.

I am not necessarily saying that my comments were entirely without fault, but I was not making them as the representative of a large corporation.

I was really shocked at the initial Twitter image and words, was even more shocked at the retaliatory Github banning for me calling them to people's attention on Reddit, and am even more shocked now that she is actually taking public digs at me personally now (I don't even have a Twitter account to respond directly with.).  I guess I expected that someone in that position would be held to a higher standard of behavior than "some random person on the Internet" (Which is essentially what I am).

While I don't doubt that many in the corporate world have the type of feelings that she expressed about her product's userbase, and would sometimes like to do all this stuff, most of the time people either reign themselves in to present a professional face publicly, or their boss reigns them in herself (or himself).

I can only assume that if no action is taken by the company, no apology is forthcoming, and my ban remains in place, that Mozilla considers this sort of conduct and retaliation appropriate conduct from their employees.

Contrary to the developer's statements, I am not attempting to "harass" this person, but rather am simply at this point trying to make sure it is enough of a thing that the higher-ups at Mozilla see it, evaluate it, and either take action or don't. At that point, people can judge for themselves whether they want to continue or begin to use that company's free software, paid products, and so on and so forth.  I don't want it to fly under the radar simply because I want to make sure I get that closure that Mozilla is either okay or not okay with this type of public employee conduct.

In any event, as it regards the URL stuff in the official Fenix Github, I clearly no longer have a public voice on there, so they'll do what they do.  My purpose with this thread was simply to call the fact that a patch existed to the attention of this fork's project maintainer, who has done some very good work in a very short time, and as a reference for anyone who may decide to use his early work as a basis for their own Fenix fork (The current project maintainer of this fork in the early going indicated that this was essentially proof of concept and not something he planned to personally maintain long-term, although he does seem open to the possibility that he may wind up at the center of a fork with some volunteers who ask him to continue and offer their services, or some group of people picking it up and using his early work and/or asking him to join their hypothetical efforts.).

I can't contribute my own code because I am not a coder, so I wanted to tie information together, at least, as a very small way of contributing to a potential fork effort. :)

Since a continuously maintained fork is probably unlikely (Though not impossible), and just because I think it's a good thing to have in a browser in any event, I would love to see the official Fenix team include the patch in their official builds.  It would be good for the Internet.

For me personally, though, the two issues are not linked.  That is to say, my posts that called attention to the stuff on Twitter were not motivated by them not acting on my feature request or their possible decision to decline to include it.  It's their browser and they can do what they want, and we can all decide whether to use it or not to use it.

I think not viewing the two matters as linked is part of why I found being banned on GitHub so shocking.  However, the Fenix developers may view them as linked.

I do want to apologize for any possible implication that @hakkikaancaliskan 's patch was not sound. That is not at all the message I was trying to send. In fact, I was very happy to see the patch, which is why I was calling attention to it in the first place.  It's just that I can't read code, and perhaps was somewhat over cautiously including a disclaimer for that reason when passing it along.

I really do very much hope that the patch gets included in the official Fenix (Though I have my doubts that it will be given the attitudes of some of their official developers.  It's not impossible, though.) and, if not, I would love to see any forks that may arise make use of it.  Actually, it would be even better to have the full URLs as the default in either or both the official browser and a potential fork for principles reasons I went into on the Mozilla repo before my ban (They may still be up if someone is actually getting this far in this post and saying "I don't think this was long enough, where can I find more written rambling from this random Internet guy?" :) ), *but* I would be reasonably satisfied with an option for those of us who want it to enable it on our personal browser iterations.  One has to take what one can get. I mentioned it as a potential compromise in my very first post to that thread.

Interestingly, during this time, I have been testing other Android browsers, and I did notice that DuckDuckGo's Android browser includes full URLs including protocol by default, which is very laudable.  Since they only offer privacy blockers (That often block ads) and not a a true ad/content-blocker, and offer no extensions, probably means that it isn't going to be my new default Android browser, but I think they deserve kudos for swimming against the tide on that one. :)

Finally, to answer your question, I guess I'm holding up alright. It's important to keep these things in perspective

**TL;DR**: I'm banned on the Mozilla Github.  Still hoping the patch (Which is not my work but is a response to my feature request from before I was banned) gets included in official, but am not sure it will. If it doesn't, I'd love to see it make its way into a fork.  I appreciate the effort of the patch creator in making this happen. Thank you.

**Comment by interfect** [comment]
Please don't get in fights with Mozilla folks. And maybe take a step back and read their code of conduct again? It's easy to not feel like you're harassing anyone, you're just advocating for your issue, but if on the other side it feels like getting harassed, there's unfortunately not much difference.

I can't promise to enforce Mozilla's code of conduct, but I urge everyone to abide by it, especially their point under "Be Respectful":

> Take responsibility for your impact and your mistakes – if someone says they have been harmed through your words or actions, listen carefully, apologize sincerely, and correct the behavior going forward.

A lot of software and technical folks love ethical theories where what matters is what you're allowed to do and what you mean to do; those are the concepts we deal with all the time and try to bridge in our work. But sometimes we instead need to concentrate on how our actions affect other people, even if the resulting effects are unintentional or otherwise shouldn't have happened. Consequentialist ethics is in.

Or maybe the Fenx devs are just being mean. I certainly can't say.



Controversy that got you banned from Mozilla's repos and possibly removed FrostedIce339 from Github notwithstanding, the linked patch seems fine and merges cleanly, so in it goes.
